packages/components/nodes/vectorstores/Couchbase/Couchbase.ts

```diff
@@ -0,0 +1,216 @@
+                type: 'string',
+                default: 'text',
+                additionalParams: true,
+                optional: true
+            },
+            {
+                label: 'Embedded Field',
+                name: 'embeddingKey',
+                description: 'Name of the field (column) that contains the Embedding',
+                type: 'string',
+                default: 'embedding',
+                additionalParams: true,
+                optional: true
+            },
+            {
+                label: 'Top K',
+                name: 'topK',
+                description: 'Number of top results to fetch. Default to 4',
+                placeholder: '4',
+                type: 'number',
+                additionalParams: true,
+                optional: true
+            }
+        ]
+        this.outputs = [
+            {
+                label: 'Couchbase Retriever',
+                name: 'retriever',
+                baseClasses: this.baseClasses
+            },
+            {
+                label: 'Couchbase Vector Store',
+                name: 'vectorStore',
+                baseClasses: [this.type, ...getBaseClasses(CouchbaseVectorStore)]
+            }
+        ]
+    }
+
+    //@ts-ignore
+    vectorStoreMethods = {
+        async upsert(nodeData: INodeData, options: ICommonObject): Promise<Partial<IndexingResult>> {
+            const credentialData = await getCredentialData(nodeData.credential ?? '', options)
+            const bucketName = nodeData.inputs?.bucketName as string
+            const scopeName = nodeData.inputs?.scopeName as string
+            const collectionName = nodeData.inputs?.collectionName as string
+            const indexName = nodeData.inputs?.indexName as string
+            let textKey = nodeData.inputs?.textKey as string
+            let embeddingKey = nodeData.inputs?.embeddingKey as string
+            const embeddings = nodeData.inputs?.embeddings as Embeddings
+
+            let connectionString = getCredentialParam('connectionString', credentialData, nodeData)
+            let databaseUsername = getCredentialParam('username', credentialData, nodeData)
+            let databasePassword = getCredentialParam('password', credentialData, nodeData)
+
+            const docs = nodeData.inputs?.document as Document[]
+
+            const flattenDocs = docs && docs.length ? flatten(docs) : []
+            const finalDocs = []
+            for (let i = 0; i < flattenDocs.length; i += 1) {
+                if (flattenDocs[i] && flattenDocs[i].pageContent) {
+                    const document = new Document(flattenDocs[i])
+                    finalDocs.push(document)
+                }
+            }
+
+            const couchbaseClient = await Cluster.connect(connectionString, {
+                username: databaseUsername,
+                password: databasePassword,
+                configProfile: 'wanDevelopment'
```

Comment:
what is `configProfile`? just trying to understand is it a must to hardcode `wanDevelopment` as the value?

Reply:
From Couchbase Node.js SDK 4.2 onwards, we have introduced a wanDevelopment Configuration Profile, which provides pre-configured timeout settings suitable for working in high latency environments.
Also this is the recommended way to connect to a cluster as per: https://docs.couchbase.com/nodejs-sdk/current/hello-world/start-using-sdk.html#connect

Reply:
appreciate the explanation!